packages/commerce-sdk-react/src/components/ShopperExperience/Page/index.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useContext, useEffect, useState} from 'react'
+import {Helmet} from 'react-helmet'
+import {Component as ComponentType, Region as RegionType, Page as PageType} from '../types'
+import {Region} from '../Region'
+
+type ComponentMap = {
+    [typeId: string]: React.ComponentType<ComponentType & unknown>
+}
+
+interface PageProps extends React.ComponentProps<'div'> {
+    page: PageType
+    components: ComponentMap
+}
+
+type PageContextValue =
+    | {
+          components: ComponentMap
+      }
+    | undefined
+
+// This context will hold the component map as well as any other future context.
+export const PageContext = React.createContext(undefined as PageContextValue)
```

Comment:
Another option without the need to us `as`.

```
export const PageContext = React.createContext<PageContextValue | undefined>(undefined)
```

Reply:
The `as PageContextValue `  was a relic from a previous type definition. It's actually not required anymore since the PageContextValue can be undefined.  I've removed the "as PageContextValue" part.